File: algorithms/504. Base 7/solution.py
class Solution(object):
    def convertToBase7(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num < 0:
            return '-' + self.convertToBase7(-num)
        res = ''
        while num:
            res = str(num % 7) + res
            num //= 7
        return res or '0'

File: algorithms/504. Base 7/solution.cc
class Solution {
public:
    string convertToBase7(int num) {
        if ( num < 0 ) {
            return "-" + convertToBase7(-num);
        }
        string result = "";
        while ( num ) {
            result = to_string(num % 7) + result;
            num /= 7;
        }
        return result == "" ? "0" : result;
    }
};

// 241 / 241 test cases passed.
// Status: Accepted
// Runtime: 3 ms
